src/test/java/io/vertx/core/eventbus/ClusteredEventBusTest.java

```diff
@@ -583,4 +587,72 @@ public void handle(Message<Object> msg) {
 
     await();
   }
+
+  @Test
+  public void testRejectedClusterSerializableNotSent() {
+    testRejectedNotSent(SomeClusterSerializableObject.class, new SomeClusterSerializableObject("bar"));
+  }
+
+  @Test
+  public void testRejectedClusterSerializableImplNotSent() {
+    testRejectedNotSent(SomeClusterSerializableImplObject.class, new SomeClusterSerializableImplObject("bar"));
+  }
+
+  @Test
+  public void testRejectedSerializableNotSent() {
+    testRejectedNotSent(SomeSerializableObject.class, new SomeSerializableObject("bar"));
+  }
+
+  private <T> void testRejectedNotSent(Class<T> clazz, T message) {
+    startNodes(2);
+    vertices[0].eventBus()
+      .clusterSerializableChecker(s -> Boolean.FALSE)
+      .serializableChecker(s -> Boolean.FALSE);
+    vertices[1].eventBus().consumer("foo", msg -> fail()).completionHandler(onSuccess(reg -> {
+      try {
+        vertices[0].eventBus().send("foo", message);
+        fail();
+      } catch (IllegalArgumentException e) {
```

Comment:
here we expect the error to be asynchronous and fail the returned future instead of being synchronous

Reply:
This is not in the scope of this PR. Failure for missing codecs have always worked like this.